src/chttpd.erl

```diff
@@ -195,30 +191,46 @@ handle_request_int(MochiReq) ->
         Other -> Other
     end,
 
-    HttpReq = #httpd{
+    Nonce = couch_util:to_hex(crypto:rand_bytes(5)),
+
+    HttpReq0 = #httpd{
         mochi_req = MochiReq,
+        begin_ts = Begin,
+        peer = Peer,
+        original_method = Method1,
+        nonce = Nonce,
         method = Method,
         path_parts = [list_to_binary(chttpd:unquote(Part))
                 || Part <- string:tokens(Path, "/")]
     },
 
+    {ok, HttpReq} = chttpd_plugin:before_request(HttpReq0),
+
+    HandlerKey =
```

Comment:
this change appears unrelated to the PR at hand. I understand the desire to clean up code but mixing in unrelated work makes reviews harder.